### PR TITLE
#177: RasterByte skipping bytes provider

### PR DESCRIPTION
**Pull request by dromagnoli**

### CODE REVIEW

File: plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/AbstractScanlineProvider.java [hunk 1]
```diff
@@ -96,5 +97,25 @@ public void writeToPngRaw(byte[] raw) {
...
+     **/
+    public static int computePixelStride(Raster raster, int[] expectedPixelStrides, boolean hasAlpha) {
+        int pixelStride = ((ComponentSampleModel) raster.getSampleModel()).getPixelStride();
+        if (raster.getNumBands() != pixelStride) {
+            return pixelStride;
+        }
+        return (expectedPixelStrides.length == 2 && hasAlpha) ? expectedPixelStrides[1] : expectedPixelStrides[0];
+    }
+
+    public static int computePixelStride(Raster raster, int[] expectedPixelStrides) {
```
Comment (aaime) [inline plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/AbstractScanlineProvider.java]: Public method without comment

File: plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/RasterByteABGRProvider.java [hunk 2]
```diff
@@ -67,7 +60,7 @@ public void next(final byte[] row, final int offset, final int length) {
                     // row[i+2] = B
                     // row[i+1] = G
                     // row[i+0] = R
-                    row[i + bandOffsets[j]] = bytes[bytesIdx + j];
+                    row[i + j] = bytes[bytesIdx + bandOffsets[j]];
```
Comment (aaime) [inline plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/RasterByteABGRProvider.java]: Confused about this change, how was it working before?

File: plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/AbstractScanlineProvider.java (hunk 1 above)
Comment (dromagnoli) [reply plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/AbstractScanlineProvider.java]: Thx @aaime. comment added

File: plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/RasterByteABGRProvider.java (hunk 2 above)
Comment (aaime) [reply plugin/png/src/main/java/it/geosolutions/imageio/plugins/png/RasterByteABGRProvider.java]: Good thanks

**Comment by aaime** [comment]
+1

**Pull request merged**